File: problems/53-maximum-subarray/maximum-subarray.kt
class Solution {
    fun maxSubArray(nums: IntArray): Int {
        var cur = 0
        var ans = nums[0]

        for(num in nums){
            cur += num
            ans = maxOf(ans, cur)
            if(cur < 0) cur = 0
        }

        return ans
    }
}

File: problems/53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maks = nums[0], cur = nums[0];
        for(int i = 1; i< size(nums); i++){
            int x = nums[i];
            cur = max(x, cur + x);
            maks = cur > maks ? cur : maks;
        }
        return maks;
    }
};
